wire/service.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper"
+	"github.com/pkg/errors"
+)
+
+// Constants representing required commands.
+const (
+	IsMasterCommand   = "isMaster"
+	WhatsMyURICommand = "whatsmyuri"
+	// Dumb, but the shell sends commands with different casing
+	BuildInfoCommand               = "buildInfo"
+	BuildinfoCommand               = "buildinfo"
+	GetLogCommand                  = "getLog"
+	GetFreeMonitoringStatusCommand = "getFreeMonitoringStatus"
+	ReplSetGetStatusCommand        = "replSetGetStatus"
+	ListCollectionsCommand         = "listCollections"
+)
+
+type service struct {
+	*mrpc.Service
+	manager jasper.Manager
+}
+
+// StartService wraps an existing Jasper manager in a mongo wire protocol
+// service and starts it. The  caller is responsible for closing the connection
+// using the returned jasper.CloseFunc.
+func StartService(ctx context.Context, m jasper.Manager, addr net.Addr) (jasper.CloseFunc, error) { //nolint: interfacer
+	host, p, err := net.SplitHostPort(addr.String())
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid address")
+	}
+	port, err := strconv.Atoi(p)
+	if err != nil {
+		return nil, errors.Wrap(err, "port is not a number")
+	}
+
+	svc := &service{
+		Service: mrpc.NewService(host, port),
+		manager: m,
+	}
+	if err := svc.registerHandlers(); err != nil {
+		return nil, errors.Wrap(err, "error registering handlers")
+	}
+
+	cctx, ccancel := context.WithCancel(context.Background())
+	go func() {
+		grip.Notice(svc.Run(cctx))
+	}()
+
+	return func() error { ccancel(); return nil }, nil
+}
+
+func (s *service) isMaster(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	resp, err := makeErrorResponse(true, nil).Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), IsMasterCommand)
+		return
+	}
+	writeResponse(ctx, w, resp, IsMasterCommand)
+}
+
+func (s *service) whatsMyURI(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	resp, err := makeWhatsMyURIResponse(s.Address()).Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), WhatsMyURICommand)
+		return
+	}
+	writeResponse(ctx, w, resp, WhatsMyURICommand)
+}
+
+func (s *service) buildInfo(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	// resp := birch.NewDocument(birch.EC.String("version", "0.0.0"))
+	// writeSuccessResponse(w, resp, BuildInfoCommand)
+	resp, err := makeBuildInfoResponse("0.0.0").Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), BuildInfoCommand)
+		return
+	}
+	writeResponse(ctx, w, resp, BuildInfoCommand)
+}
+
+func (s *service) getLog(ctx context.Context, w io.Writer, msg mongowire.Message) {
+	resp, err := makeGetLogResponse([]string{}).Message()
+	if err != nil {
+		writeErrorResponse(ctx, w, errors.Wrap(err, "could not make response"), GetLogCommand)
+		return
+	}
+	writeResponse(ctx, w, resp, GetLogCommand)
```

Comment:
as in gimlet, does it make sense to put the response writing commands into mrpc?